.githooks/pre-commit

```diff
@@ -0,0 +1,24 @@
+#!/bin/sh
+
+set -eu -o pipefail
+
+echo "Pre-commit hook running"
+
+echo "Running gofmt..."
+gofmt -w -s .
+
+echo "Running go mod tidy..."
+go mod tidy
+
+if [ ! command -v golangci-lint >/dev/null 2>&1 ]; then
+    echo "golangci-lint not installed or available in the PATH" >&2
+    echo "please check https://github.com/golangci/golangci-lint" >&2
+    exit 1
+fi
+echo "Running golangci-lint..."
+golangci-lint run
+
+echo "Running go vet..."
```

Comment:
Sorry, I forgot this earlier. Should we add `go test ./...` too or is it too much?

Reply:
Maybe not, otherwise the commits may take too long to run.

Reply:
too much bruhh, commit karne mein jeevan nikal jayega